pkg/pillar/devicenetwork/dhcpcd.go

```diff
@@ -19,29 +17,43 @@ import (
 	"strings"
 	"syscall"
 	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	log "github.com/sirupsen/logrus"
 )
 
 // UpdateDhcpClient starts/modifies/deletes dhcpcd per interface
 // Returns an ifname and error if an interface does not yet exist
-func UpdateDhcpClient(newConfig, oldConfig types.DevicePortConfig) (string, error) {
+func UpdateDhcpClient(newConfig, oldConfig types.DevicePortConfig) types.IntfStatusMap {
 
 	// Look for adds or changes
 	log.Infof("updateDhcpClient: new %v old %v\n",
 		newConfig, oldConfig)
 	// Dry-run to see if we need to ask for retry. Don't change anything
+	intfStatusMap := *types.NewIntfStatusMap()
 	for _, newU := range newConfig.Ports {
 		oldU := lookupOnIfname(oldConfig, newU.IfName)
 		if oldU == nil || oldU.Dhcp == types.DT_NONE {
 			log.Infof("updateDhcpClient: new %s dryrun", newU.IfName)
 			err := doDhcpClientActivate(newU, true)
 			if err != nil {
-				return newU.IfName, err
+				log.Errorf("UpdateDhcpClient: Failed to activate Dhcp Client. "+
+					"err: %s", err)
+				intfStatusMap.RecordFailure(newU.IfName, err.Error())
+			} else {
+				intfStatusMap.RecordSuccess(newU.IfName)
```

Comment:
Sorry, but a "success" means it managed to connect to the controller. So you can't reuse this since the user will be utterly confused by it.

Reply:
Hmm.. Yeah.. Then we should only record failure here, but not success.. Success should be recorded only when the connectivity test passes on the interface